dipy/sims/voxel.py

```diff
@@ -290,8 +295,273 @@ def multi_tensor(gtab, mevals, S0=100, angles=[(0, 0), (90, 0)],
+    taylor expansion approximation are neglected.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from dipy.sims.voxel import multi_tensor_dki
+    >>> from dipy.data import get_data
+    >>> from dipy.core.gradients import gradient_table
+    >>> from dipy.io.gradients import read_bvals_bvecs
+    >>> fimg, fbvals, fbvecs = get_data('small_64D')
+    >>> bvals, bvecs = read_bvals_bvecs(fbvals, fbvecs)
+    >>> bvals_2s = np.concatenate((bvals, bvals * 2), axis=0)
+    >>> bvecs_2s = np.concatenate((bvecs, bvecs), axis=0)
+    >>> gtab = gradient_table(bvals_2s, bvecs_2s)
+    >>> mevals = np.array([[0.00099, 0, 0],[0.00226, 0.00087, 0.00087]])
+    >>> S, dt, kt =  multi_tensor_dki(gtab, mevals)
+
+    References
+    ----------
+    .. [1] R. Neto Henriques et al., "Exploring the 3D geometry of the
+           diffusion kurtosis tensor - Impact on the development of robust
+           tractography procedures and novel biomarkers", NeuroImage (2015)
+           111, 85-99.
+    """
+
+    if np.round(np.sum(fractions), 2) != 100.0:
+        raise ValueError('Fractions should sum to 100')
+
+    fractions = [f / 100. for f in fractions]
+
+    S = np.zeros(len(gtab.bvals))
+
+    sticks = _check_directions(angles)
+
+    # computing a 3D matrix containing the individual DT components
+    mD = np.zeros((len(fractions), 3, 3))
+    for i in range(len(fractions)):
+        R = all_tensor_evecs(sticks[i])
+        mD[i] = dot(dot(R, np.diag(mevals[i])), R.T)
+
+    # compute voxel's DT
+    D = np.zeros((3, 3))
+    for i in range(len(fractions)):
+        D = D + fractions[i]*mD[i]
+    dt = np.array([D[0][0], D[1][1], D[2][2], D[0][1], D[0][2], D[1][2]])
+
+    # compute voxel's KT
+    kt = np.zeros((15))
+    kt[0] = compute_Wijkl(mD, fractions, 0, 0, 0, 0)
+    kt[1] = compute_Wijkl(mD, fractions, 1, 1, 1, 1)
+    kt[2] = compute_Wijkl(mD, fractions, 2, 2, 2, 2)
+    kt[3] = compute_Wijkl(mD, fractions, 0, 0, 0, 1)
+    kt[4] = compute_Wijkl(mD, fractions, 0, 0, 0, 2)
+    kt[5] = compute_Wijkl(mD, fractions, 0, 1, 1, 1)
+    kt[6] = compute_Wijkl(mD, fractions, 1, 1, 1, 2)
+    kt[7] = compute_Wijkl(mD, fractions, 0, 2, 2, 2)
+    kt[8] = compute_Wijkl(mD, fractions, 1, 2, 2, 2)
+    kt[9] = compute_Wijkl(mD, fractions, 0, 0, 1, 1)
+    kt[10] = compute_Wijkl(mD, fractions, 0, 0, 2, 2)
+    kt[11] = compute_Wijkl(mD, fractions, 1, 1, 2, 2)
+    kt[12] = compute_Wijkl(mD, fractions, 0, 0, 1, 2)
+    kt[13] = compute_Wijkl(mD, fractions, 0, 1, 1, 2)
+    kt[14] = compute_Wijkl(mD, fractions, 0, 1, 2, 2)
+
+    # compute S based on the DT and KT
+    S = single_diffkurt_tensors(gtab, dt, kt, S0, snr)
+
+    return S, dt, kt
+
+
+def compute_Wijkl(Dc, frac, ind_i, ind_j, ind_k, ind_l):
+    r""" Computes the diffusion kurtosis tensor element (with indexes i, j, k
+    and l) based on the individual diffusion tensor components of a
+    multicompartmental model.
+
+    Parameters
+    -----------
+    Dc : (K,3,3) ndarray
```

Comment:
We should probably change this variable name to something more informative. How about `diffusion_tensors`?